golem-worker-service-base/src/api/register_api_definition_api.rs

```diff
@@ -163,17 +163,27 @@ pub struct HttpApiDefinitionResponseData {
     pub created_at: Option<chrono::DateTime<chrono::Utc>>,
 }
 
-impl<Namespace> From<CompiledHttpApiDefinition<Namespace>> for HttpApiDefinitionResponseData {
-    fn from(value: CompiledHttpApiDefinition<Namespace>) -> Self {
-        let routes = value.routes.into_iter().map(|route| route.into()).collect();
+impl<Namespace> TryFrom<CompiledHttpApiDefinition<Namespace>> for HttpApiDefinitionResponseData {
+    type Error = String;
+    fn try_from(value: CompiledHttpApiDefinition<Namespace>) -> Result<Self, String> {
+        let mut routes = vec![];
 
-        Self {
+        for route in value.routes {
+            // We shouldn't expose auth call back binding to users
+            // as it is giving away the internal details of the call back system that enables security.
+            if !route.binding.is_security_binding() {
+                let v = RouteWithTypeInfo::try_from(route)?;
+                routes.push(v);
+            }
+        }
```

Comment:
In case this if less condition taken off by anyone, tests will fail with error messages, that the auth call back endpoint shouldn't be exposed.